### DISCUSSION TITLE
disabling caching in tests?

### DISCUSSION THREAD

**Discussion by adjstreams** [arqo123/speedgoose#53] [General]
I've got set of methods I used during beforeAll, afterAll, afterEach, etc, which sets up an in-memory mongo db, clears the collections, and clears the db.

In some of my queries I'm using .lean().cacheQuery(), and I believe what is happening, is that even though the collection is cleared after each test, the cachedQuery still finds the record. I can get around this by changing the values of the document being added in each test so that the Id it's finding doesn't match a previous one, but I wondered if there's a way to disable caching in all or certain tests?

Here's my current setup methods that use in my test suites:

```typescript
import mongoose from "mongoose";
import { MongoMemoryServer } from "mongodb-memory-server";
import { applySpeedGooseCacheLayer, SharedCacheStrategies } from "speedgoose";
let mongo: MongoMemoryServer;

export const connectDB = async (): Promise<void> => {
  mongo = await MongoMemoryServer.create();
  const uri = mongo.getUri();
  await mongoose.connect(uri, {
    useNewUrlParser: true,
    useUnifiedTopology: true,
    useFindAndModify: false,
  });

  applySpeedGooseCacheLayer(mongoose, {
    sharedCacheStrategy: SharedCacheStrategies.IN_MEMORY,
  });
};

export const dropDB = async (): Promise<void> => {
  if (mongo) {
    await mongoose.connection.dropDatabase();
    await mongoose.connection.close();
    await mongo.stop();
  }
};

export const dropCollections = async (): Promise<void> => {
  if (mongo) {
    const collections = await mongoose.connection.db.collections();
    for (const collection of collections) {
      await collection.deleteMany({});
    }
  }
};

```

**Comment by arqo123** [comment]
Sure! Just separate your config according to the environment
![image](https://user-images.githubusercontent.com/26055043/236744431-8f6b0575-6182-4161-8e09-03beaecf4775.png)


So your test config should be like 

```
 applySpeedGooseCacheLayer(mongoose, {
    sharedCacheStrategy: SharedCacheStrategies.IN_MEMORY,
    enabled: false
  });
  ```
  
  If you would like to enable caching for the given test you can prepare some function to prepare your test setup, and run it before running functions. 
  
  So you can do it like
  
  
![image](https://user-images.githubusercontent.com/26055043/236745564-a62cb97f-ab51-4a8a-ba7f-16761f1d28b5.png)
